### PR TITLE
chore: upgrade agp to 8.1

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

Upgrade AGP from 7.4.1 to 8.1 to be able to use [baseline profiler](https://developer.android.com/topic/performance/baselineprofiles/overview)

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1208](https://10.10.124.134/job/AR-build-pipeline/1208/) **failed**.

**Comment by AndroidBob** [comment]
Build [1212](https://10.10.124.134/job/AR-build-pipeline/1212/) **failed**.

**Review by saleniuk — COMMENTED** [review]
question: have you checked how it works with proguard? In 8.1 default mode for the R8 is "fullMode", so it can perform more aggressive optimizations (https://r8.googlesource.com/r8/+/refs/heads/master/compatibility-faq.md#r8-full-mode).

Not sure if you know but you can use the AGP Upgrade Assistant to see what needs to be changed or verified and get some suggestions (in AS: Tools -> AGP Upgrade Assistant) 😄

**Comment by ohassine** [comment]
> question: have you checked how it works with proguard? In 8.1 default mode for the R8 is "fullMode", so it can perform more aggressive optimizations (https://r8.googlesource.com/r8/+/refs/heads/master/compatibility-faq.md#r8-full-mode).

Nothing to do, R8 works with the rules of Proguard.

> Not sure if you know but you can use the AGP Upgrade Assistant to see what needs to be changed or verified and get some suggestions (in AS: Tools -> AGP Upgrade Assistant) 😄

That tool could work for small projects, in our case it could mess up things by adding some line in wrong places/gradle files.

**Comment by AndroidBob** [comment]
Build [395](https://10.10.124.134/job/AR-build-pipeline/395/) **failed**.

**Comment by AndroidBob** [comment]
Build [411](https://10.10.124.134/job/AR-build-pipeline/411/) **failed**.

**Comment by AndroidBob** [comment]
Build [413](https://10.10.124.134/job/AR-build-pipeline/413/) **failed**.

**Comment by AndroidBob** [comment]
Build [417](https://10.10.124.134/job/AR-build-pipeline/417/) **failed**.

**Comment by AndroidBob** [comment]
Build [422](https://10.10.124.134/job/AR-build-pipeline/422/) **failed**.

File: buildSrc/src/main/kotlin/scripts/infrastructure.gradle.kts [hunk 1]
```diff
@@ -40,7 +41,13 @@ tasks.register("runUnitTests") {
 
 tasks.register("runAcceptanceTests") {
     description = "Runs all Acceptance Tests in the connected device."
-    dependsOn(":app:connected${Default.BUILD_FLAVOR.capitalize()}DebugAndroidTest")
+    dependsOn(
+        ":app:connected${
+            Default.BUILD_FLAVOR.replaceFirstChar {
+                if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
+            }
```
Comment (MohamadJaara) [inline buildSrc/src/main/kotlin/scripts/infrastructure.gradle.kts]: suggestion: why not just make the first char a upper case 

```suggestion
            Default.BUILD_FLAVOR.replaceFirstChar(Char::titlecaseChar)
```

File: buildSrc/src/main/kotlin/scripts/quality.gradle.kts [hunk 2]
```diff
@@ -90,10 +91,9 @@ val jacocoReport by tasks.registering(JacocoReport::class) {
     group = "Quality"
     description = "Reports code coverage on tests within the Wire Android codebase"
     val buildVariant = "devDebug" // It's not necessary to run unit tests on every variant so we default to "devDebug"
-    dependsOn("test${buildVariant.capitalize()}UnitTest")
+    dependsOn("test${buildVariant.replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }}UnitTest")
```
Comment (MohamadJaara) [inline buildSrc/src/main/kotlin/scripts/quality.gradle.kts]: same

File: buildSrc/src/main/kotlin/scripts/infrastructure.gradle.kts (hunk 1 above)
Comment (MohamadJaara) [reply buildSrc/src/main/kotlin/scripts/infrastructure.gradle.kts]: and this change my break Jenkins

Comment (ohassine) [reply buildSrc/src/main/kotlin/scripts/infrastructure.gradle.kts]: could be let's try it

**Comment by AndroidBob** [comment]
Build [451](https://10.10.124.134/job/AR-build-pipeline/451/) **failed**.

**Comment by AndroidBob** [comment]
Build [452](https://10.10.124.134/job/AR-build-pipeline/452/) **failed**.

**Comment by AndroidBob** [comment]
Build [462](https://10.10.124.134/job/AR-build-pipeline/462/) **failed**.

**Comment by AndroidBob** [comment]
Build [464](https://10.10.124.134/job/AR-build-pipeline/464/) **failed**.

**Comment by AndroidBob** [comment]
Build [489](https://10.10.124.134/job/AR-build-pipeline/489/) **failed**.

**Comment by AndroidBob** [comment]
Build [492](https://10.10.124.134/job/AR-build-pipeline/492/) **failed**.

**Comment by AndroidBob** [comment]
Build [499](https://10.10.124.134/job/AR-build-pipeline/499/) **failed**.

**Comment by AndroidBob** [comment]
Build [539](https://10.10.124.134/job/AR-build-pipeline/539/) **failed**.

**Comment by AndroidBob** [comment]
Build [551](https://10.10.124.134/job/AR-build-pipeline/551/) **failed**.

**Comment by AndroidBob** [comment]
Build [571](https://10.10.124.134/job/AR-build-pipeline/571/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7495043)(7495043)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7495043)(7495043)-staging-compat.apk)

**Review by vitorhugods — APPROVED** [review]

File: app/proguard-rules.pro [hunk 3]
```diff
@@ -40,3 +40,8 @@
 #-keepattributes InnerClasses # Needed for `getDeclaredClasses`.
 
 -keepnames class * extends com.ramcosta.composedestinations.spec.Route
+
+-dontwarn java.awt.Component
+-dontwarn java.awt.GraphicsEnvironment
+-dontwarn java.awt.HeadlessException
+-dontwarn java.awt.Window
```
Comment (MohamadJaara) [inline app/proguard-rules.pro]: can you please add a comment on why those rules are added

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [581](https://10.10.124.134/job/AR-build-pipeline/581/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7495635)(7495635)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7495635)(7495635)-staging-compat.apk)

**Pull request merged**